src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleDynamicAssociationsController.cs

```diff
@@ -17,22 +17,26 @@ namespace VirtoCommerce.CatalogModule.Web.Controllers.Api
     [Authorize]
     public class CatalogModuleDynamicAssociationsController : Controller
     {
-        private readonly IDynamicAssociationSearchService _searchService;
-        private readonly IDynamicAssociationService _service;
+        private readonly IDynamicAssociationSearchService _dynamicAssociationSearchService;
+        private readonly IDynamicAssociationService _dynamicAssociationService;
         private readonly IAuthorizationService _authorizationService;
 
-        public CatalogModuleDynamicAssociationsController(IDynamicAssociationSearchService searchService, IDynamicAssociationService service, IAuthorizationService authorizationService)
+        public CatalogModuleDynamicAssociationsController(
+            IDynamicAssociationSearchService dynamicAssociationSearchService,
+            IDynamicAssociationService dynamicAssociationService,
+            IAuthorizationService authorizationService)
         {
-            _searchService = searchService;
-            _service = service;
+            _dynamicAssociationSearchService = dynamicAssociationSearchService;
+            _dynamicAssociationService = dynamicAssociationService;
             _authorizationService = authorizationService;
         }
 
         [HttpPost]
         [Route("search")]
+        [Authorize(ModuleConstants.Security.Permissions.Read)]
         public async Task<ActionResult<DynamicAssociationSearchResult>> SearchAssociation([FromBody] DynamicAssociationSearchCriteria criteria)
         {
-            var result = await _searchService.SearchDynamicAssociationsAsync(criteria);
+            var result = await _dynamicAssociationSearchService.SearchDynamicAssociationsAsync(criteria);
```

Comment:
Need to add authorization for `DynamicAssociationSearchCriteria` object here. We need to filter all stores with non available catalogs